src/CXX/Python3/cxx_extensions.cxx

```diff
@@ -731,6 +731,26 @@ PythonType &PythonType::set_methods( PyMethodDef *methods )
     return *this;
 }
 
+PythonType &PythonType::set_tp_descr_get( PyObject *(*tp_descr_get)( PyObject *self, PyObject *obj, PyObject *type ) )
```

Comment:
Some Linux distributions (e.g. Debian, Fedora, ...) have the policy to use a system-package whenever possible. In this case python3-cxx will be used instead of the copied PyCXX code of FreeCAD. Now this change will break API compatibility because upstream doesn't provide this function.

A workaround could be to implement a helper class in FreeCADBase that adds further slots of the type structure.

For example:
```
class BaseExport PythonTypeExt
{
public:
    PythonTypeExt(Py::PythonType&);
    PythonType &set_tp_descr_get( PyObject *(*tp_descr_get)( PyObject *self, PyObject *obj, PyObject *type ) );
    PythonType &set_tp_descr_set( int (*tp_descr_set)( PyObject *self, PyObject *obj, PyObject *value ) );

private:
    PythonType& pytype;
};
```
and
```
PythonTypeExt::PythonTypeExt(Py::PythonType& type)
    : pytype(type)
{
}

PythonType &PythonTypeExt::set_tp_descr_get( PyObject *(*tp_descr_get)( PyObject *self, PyObject *obj, PyObject *type ) )
{
    pytype.type_object()->tp_descr_get = tp_descr_get;
    return pytype;
}

PythonType &PythonTypeExt::set_tp_descr_set( int (*tp_descr_set)( PyObject *self, PyObject *obj, PyObject *value ) )
{
    pytype.type_object()->tp_descr_set = tp_descr_set;
    return pytype;
}
```

In CommandActionPy you can use it this way:
```
void CommandActionPy::init_type()
{
...
    PythonTypeExt ext(behaviors());
    ext.set_tp_descr_get(&CommandActionPy::descriptorGetter);
    ext.set_tp_descr_set(&CommandActionPy::descriptorSetter);
}
```

Reply:
Ok, I will modify it as you suggest.